server/index.js

```diff
@@ -48,14 +48,32 @@ async function setupApp() {
 
   });
 
+  app.post('/users', async (req, res) => {
+    try {
+      const { name, email, password, birthdate, notification_preference, biography, terms_agreed, newsletter_subscribed } = req.body;
+      const user = await pgClient.query(`SELECT * FROM  messages WHERE email = $1`, [email]);
+      if(user.length !== 0){
+        return res.json({ error: 'email already exists' });
+      }
+      if (user.password.length < 6){
+        return res.status(401).json({ message: 'password has to be more than 6 characters' });
+      }
+
+      const userData = await pgClient.query(`INSERT INTO users (name, email, birthdate, notification_preference, terms_agreed, newsletter_subscribed) VALUES ($1, $2, $3, $4, $5, $6) RETURNING *`, [name, email, password, birthdate, notification_preference, biography, terms_agreed, newsletter_subscribed ]);
+      return res.json(userData);
+    } catch(err){
+      console.error(err.message)
```

Comment:
We should make sure to also send a res  in case of error.

e.g.

```
return res.status(500).send({ error: 'Internal Server Error' });
```

Reply:
Json isn't preferred?

Reply:
@Pybite json vs send? Either one works same way.

Reply:
@Pybite send() calls json() behind the scenes:

https://github.com/expressjs/express/blob/4cf7eed927d3ccd3f1d0c9a14d562ec0a1635e86/lib/response.js#L162

Reply:
Ahh thank you ! I surely will read up on it right now